django_celery_results/backends/database.py

```diff
@@ -41,8 +43,13 @@ def _get_task_meta_for(self, task_id):
         obj = self.TaskModel._default_manager.get_task(task_id)
         res = obj.as_dict()
         meta = self.decode_content(obj, res.pop('meta', None)) or {}
-        res.update(meta,
-                   result=self.decode_content(obj, res.get('result')))
+        result = self.decode_content(obj, res.get('result'))
+        task_args = self.decode_content(obj, res.get('task_args'))
+        task_kwargs = self.decode_content(obj, res.get('task_kwargs'))
```

Comment:
The args and kwargs encoding shouldn't match the result encoding, which is what this logic essentially does since you pass in obj and decode_content checks obj.content_encoding.  You ignore the returned content_encoding for those items above, so it's basically lost information.   Result encoding can be binary for instance, but args/kwargs may not be, and vice versa.

Reply:
I use `self.decode_content(...)` to deserialize because I used `self.encode_content(...)` to serialize. I thought that it was consistent. I don't know any other way to do it.

Reply:
Look at how the decode works.

Reply:
Look at how encode works.

Reply:
The `content_encoding` that returns the `encode_content` function only depends on `self.serializer`. Which means that it will be the same for `result`, `meta`, `task_args`, `task_kwargs` and for all others objects serialized using the `encode_content` of the same DatabaseBackend instance.

Moreover, we're also ignoring the `content_encoding` of `meta` and there is error when we deserialize it using the same `decode_content` function.

Reply:
obj.content_encoding is set according to result, but encoding doesn't take that into account, it only looks at the data for the field in question.  What you've done is made a case where the result may be binary but it encoded your args as json... but then trying to decode the args would try to decode the binary which it wasn't encoded as.  

Basically content_encoding was only ever meant for result.  If you want to encode the args/kwargs you need to make sure it always handles them the same way, otherwise there will be a lot of inconsistency.

Reply:
> obj.content_encoding is set according to result

False, it is not set according to `result`. It is set according to `self.serializer`, which is always the same.

Kombu `dumps()` function would only take `result` into account if `self.serializer` were `None` or `"raw"`. But this is not possible because celery's allowed serializers are only: json, pickle, yaml and msgpack (or custom serializer, but not None nor "raw").

Reply:
Hmm, maybe I'm reading this wrong.  What is the obj args in decode_content?  I was assuming that was the django model instance.

Reply:
I added more test where:
- result is a dict
- result is text
- result is bytes

to ensure that it works fine with all of them.

Reply:
> Hmm, maybe I'm reading this wrong. What is the obj args in decode_content? I was assuming that was the django model instance.

The `obj` argument of the `decode_content()` function is a `TaskResult` instance.

Reply:
ah, okay, then this makes much more sense, sorry about the confusion.  If all your extra tests are passing then I'm good besides @auvipy 's concern about backward compatibility.